pymaker/dss.py

```diff
@@ -384,6 +415,51 @@ def frob(self, ilk: Ilk, urn_address: Address, dink: Wad, dart: Wad, collateral_
         return Transact(self, self.web3, self.abi, self.address, self._contract,
                         'frob', [ilk.toBytes(), urn_address.address, v.address, w.address, dink.value, dart.value])
 
+    def simulate_frob(self, collateral: Collateral, address: Address, dink: Wad, dart: Wad):
```

Comment:
I'd suggest to keep this code in test module (maybe as a Collateral Test class that extends Collateral and has simulate_frob?)

Reply:
Not sure I understand; the whole motivation of this change is to allow the method to be used by other libraries.  Also, `simulate_frob` pertains to the `vat`, not a collateral type.

An alternate implementation would be to move all the assertions into `vat.frob` and run them for each `frob`.  Would that be more favorable?

Reply:
I could be wrong but from what I see simulate_frob is a test utility and won't be used in normal circumstances? If so then it should stay in a test module / test utility that can be imported by other projects and not in main Collateral implementation

Reply:
You're correct.  I moved it from `TestVat` to `Vat` so it could be used by other projects.  Would it be better to put in `conftest`?